rmw/include/rmw/rmw.h

```diff
@@ -803,6 +803,41 @@ RMW_WARN_UNUSED
 rmw_ret_t
 rmw_publisher_assert_liveliness(const rmw_publisher_t * publisher);
 
+/// Wait until all published message data is acknowledged or until the specified timeout elapses.
+/**
+ * This function waits until all published message data were acknowledged by peer node or timeout.
+ *
+ * \note This function only works effectively while QOS profile of publisher is set to RELIABLE.
+ *   Otherwise this function will immediately return RMW_RET_OK.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Maybe [1]
+ * Thread-Safe        | No
+ * Uses Atomics       | Maybe [1]
+ * Lock-Free          | Maybe [1]
+ * <i>[1] rmw implementation defined, check the implementation documentation</i>
+ *
+ * \param[in] publisher handle to the publisher that needs to wait for all acked.
+ * \param[in] wait_timeout If zero, do not block -- check only for immediately all published message
+ *   data. Else, this represents the maximum amount of time to wait for all published message data
```

Comment:
```suggestion
 * \param[in] wait_timeout represents the maximum amount of time to wait for all published message data
```

Clarifying what happens for timeout zero doesn't seem to be worth it.
The DDS `wait_for_acknowledgments()` function likely returns immediately and doesn't check anything in that case.